test/cache-fs.spec.ts

```diff
@@ -0,0 +1,64 @@
+import {EventEmitter} from 'eventemitter3';
+import * as Promise from 'bluebird';
+import {expect} from 'chai';
+import {EventsMatcher} from '../test-kit/drivers/events-matcher';
+import {SlowFs} from '../test-kit/drivers/slow-fs';
+import {FileSystem} from '../src/api';
+import {CacheFs} from '../src/cache-fs';
+import {MemoryFileSystem} from '../src/memory-fs';
+import {assertFileSystemContract} from './implementation-suite';
+
+describe(`the cache file system implementation`, () => {
+
+    assertFileSystemContract(
+        () => Promise.resolve(new CacheFs(new MemoryFileSystem())),
+        {
+            interval: 1,
+            noExtraEventsGrace: 10,
+            timeout: 30
+        }
+    );
+
+    describe(`using slow FileSystem`, () => {
+        const fileName = 'foo.txt';
+        const content = 'content';
+        const timeout = 200;
+
+        let timer;
+        let fs: FileSystem;
+        let slow: FileSystem;
+        let startTimestamp: number;
+
+        beforeEach(() => {
+            startTimestamp = Date.now();
+            slow = new SlowFs(timeout)
+            fs = new CacheFs(slow)
+        });
+
+
+        it('loads file faster after it has been saved', () => {
+            return fs.saveFile(fileName, content)
+                .then(() => fs.loadTextFile(fileName))
+                .then(() => expect(Date.now() - startTimestamp).to.be.lessThan(timeout * 2));
+        })
+
+        it('loads file faster after it has been saved from outside', () => {
```

Comment:
what is faster here?
it takes timeout * 2 which is exactly the time it would take without a cache layer